### PR TITLE
fix: metropolis step size init, coupling with odd dimensions

### LINKED ISSUE #33: Model supports only targets with even dimensions, need support for uneven dimensions
Hi Laurence Midgley,

Thank you for providing the JAX implementation of the FAB model!

I am currently trying to apply your model to targets with uneven dimensions. However, I've encountered an issue as the current implementation does not seem to support this for both the real_nvp and spline flows. As far as I know is that these flows generally support uneven dimensions.

Could you please provide guidance on how to apply FAB to problems with uneven dimensions, if possible?

Kind regards,
Johannes

### PR DESCRIPTION
**Pull request by lollcat**

### CODE REVIEW

File: experiments/config/gmm_v0.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ fab:
       tune_step_size: true
     metropolis:
       n_outer_steps: 1
-      init_step_size: 5.
+      init_step_size: 10.
```
Comment (lollcat) [inline experiments/config/gmm_v0.yaml:29]: Change so that behavior is exactly the same as before this issue (step size was being (incorrectly) multiplied by dim.

File: experiments/gmm_v0.py [hunk 2]
```diff
@@ -9,15 +9,15 @@
...
+    local = True
 
     if local:
         if "logger" in cfg.keys():
             del cfg.logger
     if cfg.training.use_64_bit:
         jax.config.update("jax_enable_x64", True)
 
-    target = GMM()
+    target = GMM(dim=3)
```
Comment (lollcat) [inline experiments/gmm_v0.py]: now this works

File: fabjax/flow/build_coupling_bijector.py [hunk 3]
```diff
@@ -43,7 +43,7 @@ def build_split_coupling_bijector(
     bijectors = []
     for swap in (True, False):
         params_after_split = dim - split_index
-        params_transformed = params_after_split if swap else split_index
```
Comment (lollcat) [inline fabjax/flow/build_coupling_bijector.py:46]: bug fix
